### PR TITLE
Clean up all remaining Lint warnings

### PR DESCRIPTION
**Pull request by martinmaly**

### CODE REVIEW

**Comment by duglin** [comment]
LGTM

**Comment by vaikas-google** [comment]
LGTM

File: broker/user_provided/controller/controller.go [hunk 1]
```diff
@@ -26,26 +26,27 @@ import (
...
 
-type Controller struct {
-	instanceMap map[string]*UserProvidedServiceInstance
+type userProvidedController struct {
+	instanceMap map[string]*userProvidedServiceInstance
 }
 
 // Verify that Controller implements the broker Controller interface.
-var _ controller.Controller = (*Controller)(nil)
+var _ controller.Controller = (*userProvidedController)(nil)
```
Comment (arschles) [inline broker/user_provided/controller/controller.go]: verifying that the `*userProvidedController` conforms to `Controller` should be left to test code. also, note that the code in `CreateController` does so anyway

File: broker/user_provided/controller/controller.go [hunk 2]
```diff
@@ -88,26 +89,26 @@ func (c *Controller) CreateServiceInstance(id string, req *sbmodel.ServiceInstan
...
-		return errors.New("Not found")
 	}
+
+	return errors.New("Not found")
 }
 
-func (c *Controller) Bind(instanceId string, bindingId string, req *sbmodel.BindingRequest) (*sbmodel.CreateServiceBindingResponse, error) {
-	cred := c.instanceMap[instanceId].Credential
+func (c *userProvidedController) Bind(instanceID string, bindingID string, req *sbmodel.BindingRequest) (*sbmodel.CreateServiceBindingResponse, error) {
+	cred := c.instanceMap[instanceID].Credential
```
Comment (arschles) [inline broker/user_provided/controller/controller.go]: since `cred` is dereferenced on the next line, it needs to be `nil`-checked

**Comment by pmorie** [comment]
LGTM

File: controller/server/third_party_service_storage.go [hunk 3]
```diff
@@ -13,19 +14,21 @@ import (
...
+// backed by Kubernetes third-party resources.
 func NewThirdPartyServiceStorage(w *watch.Watcher) ServiceStorage {
-	return &ThirdPartyServiceStorage{
+	return &thirdPartyServiceStorage{
 		watcher: w,
 	}
 }
 
-var _ ServiceStorage = (*ThirdPartyServiceStorage)(nil)
+var _ ServiceStorage = (*thirdPartyServiceStorage)(nil)
```
Comment (arschles) [inline controller/server/third_party_service_storage.go]: if this is verifying that `*ThirdPartyServiceStorage` is a `ServiceStorage`, that should be left for test code. also, `NewThirdPartyServiceStorage` accomplishes as well

**Pull request merged**